### ISSUE TITLE
We have updated instructions on how to apply contrastive search on encoder-decoder models (e.g. BART and T5).

### ISSUE THREAD

**Issue by yxuansu** [yxuansu/SimCTG#4]
If you are looking for codes of how to apply contrastive search on encoder-decoder models (e.g. BART and T5). Please find more details here (https://github.com/yxuansu/SimCTG/tree/main/SimCTGEncDec).

**Comment by Atharva-Phatak** [comment]
@yxuansu Perfect thanks for the effort. This will help me a lot. I will contact you if I have any issues :)

**Comment by yxuansu** [comment]
> @yxuansu Perfect thanks for the effort. This will help me a lot. I will contact you if I have any issues :)

You are welcome. Please free feel to contact if you meet any problem :-)

**Comment by Atharva-Phatak** [comment]
Have you used simCTG on tasks like summarization ? After reading the paper I think simCTG can be adopted to tasks like text-summarization. Since simCTG was designed for open-ended generation how can I adopt it for text summarization ?

Any tips or advices ?

**Comment by yxuansu** [comment]
> Have you used simCTG on tasks like summarization ? After reading the paper I think simCTG can be adopted to tasks like text-summarization. Since simCTG was designed for open-ended generation how can I adopt it for text summarization ?
> 
> Any tips or advices ?

Hi,

I definitely think SimCTG can be applied to tasks like summarization or translation. My advice is to follow the same procedure as described in the paper: (1) first using contrastive training to train you summarization model (e.g. BART or T5); (2) use contrastive search to generate the result.

For the contrastive training on encoder-decoder models, I recommend you to read this blog (https://zenn.dev/kwashizzz/articles/ml-simctg-contrastive-framework). They provide a good tutorial and code implementation on applying contrastive training on the T5 model. After contrastive training, you can refer to our tutorial (https://github.com/yxuansu/SimCTG/tree/main/SimCTGEncDec) on how to apply contrastive search to your trained model.

Please let me know if you have any further questions.

**Comment by Atharva-Phatak** [comment]
in your instructions do you apply diverse contrastive search ? if not any hints on how to implement it ?

**Comment by yxuansu** [comment]
> in your instructions do you apply diverse contrastive search ? if not any hints on how to implement it ?
> 
> Also what is the effect of beam_width ?

No, we do not implement the diverse contrastive search. I think you can easily adapt the code by yourself by referring to the details here (https://github.com/yxuansu/SimCTG/blob/9d294f7bd06d151f52c3d278b55cdebae0225e33/document_generation/simctg.py#L135).

Regarding the beam width, it does not affect that much as compared with \alpha, but I recommend you to try different values of beam width to see the performance difference.

**Comment by Atharva-Phatak** [comment]
I was training BART for summarization using simCTG on custom dataset, the model does not seem to improve after first epoch. I do not use external metrics such as rogue, I am using validation loss to save the best model. Any recommendations to stabilise the training and improve the performance ?
